napari/layers/volume/volume.py

```diff
@@ -150,7 +159,17 @@ def data(self, data):
         self.refresh()
 
     def _get_shape(self):
-        return self.data.shape
+        return tuple(np.multiply(self.data.shape, self.spacing))
+
+    @property
+    def spacing(self):
+        """list: Anisotropy factors to scale the volume by."""
+        return self._spacing
+
+    @spacing.setter
+    def spacing(self, spacing):
+        self._spacing = spacing
+        self.scale = [self._spacing[s] for s in [-1, -2, -3]]
```

Comment:
I *think* this is better?

```python
visible_dims = ...
self.scale = [self._spacing[s] for s in visible_dims[::-1]
```

Then you might be able to put that code in the base layer instead of Volume...?

Reply:
yeah - that will also make the integration with #451 easier as that PR creates an explicit parameter of that sort. I'll still probably keep in the volume for now though, as i'm not sure how we want to think about this sort of scaling yet - for example would be support this sort of scaling to convert things from pixels to microns?

Reply:
fixed